### PR TITLE
bugfix google button and modal windows

### PR DESCRIPTION
**Pull request by Mygal-Alex**
https://github.com/ita-social-projects/SpaceToStudy-Client/assets/101183408/163d7ca9-31a6-4a6b-b19b-b518c17e9bbc

### CODE REVIEW

File: src/containers/guest-home-page/login-dialog/LoginDialog.styles.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ const style = {
     maxWidth: { sm: 'sm', md: 'md', lg: 'lg' },
     mt: { xs: '56px', sm: 0 },
     display: 'flex',
-    justifyContent: 'space-between',
+    justifyContent: 'space-around;',
```
Comment (Tolik170) [inline src/containers/guest-home-page/login-dialog/LoginDialog.styles.js:8]: remove semicolon

File: src/containers/guest-home-page/signup-dialog/SignupDialog.jsx [hunk 2]
```diff
@@ -26,6 +26,7 @@ import tutor from '~/assets/img/signup-dialog/tutor.svg'
 import info from '~/assets/img/guest-home-page/info.svg'
 
 import { styles } from '~/containers/guest-home-page/signup-dialog/SignupDialog.styles'
+import useBreakpointsGoogle from '~/hooks/use-breakpoints-google'
```
Comment (Tolik170) [inline src/containers/guest-home-page/signup-dialog/SignupDialog.jsx:29]: organize imports:
-libs
-services, hooks, containers, components
-utils, types, styles

add space between them

**Review by ArtemHolikov — APPROVED** [review]

File: src/hooks/use-breakpoints-google.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import useMediaQuery from '@mui/material/useMediaQuery'
+
+const useBreakpointsGoogle = () => {
```
Comment (Tolik170) [inline src/hooks/use-breakpoints-google.tsx:3]: do you really need this hook, as I see, you can use our useBreakpoints for that

File: src/containers/guest-home-page/signup-dialog/SignupDialog.styles.js [hunk 4]
```diff
@@ -5,7 +5,7 @@ export const styles = {
     maxWidth: { sm: 'sm', md: 'md', lg: 'lg' },
     mt: { xs: '56px', sm: 0 },
     display: 'flex',
-    justifyContent: 'space-between',
+    justifyContent: 'space-around;',
```
Comment (Tolik170) [inline src/containers/guest-home-page/signup-dialog/SignupDialog.styles.js:8]: remove semicolon

File: src/containers/guest-home-page/signup-dialog/SignupDialog.jsx [hunk 5]
```diff
@@ -84,7 +85,24 @@ const SignupDialog = ({ type }) => {
   useEffect(() => {
     setNeedConfirmation(isDirty)
   }, [isDirty, setNeedConfirmation])
+  const breakpoints = useBreakpointsGoogle()
 
+  const { isSmallScreen, isMediumScreen, isLargeScreen, isXLargeScreen } =
```
Comment (Tolik170) [inline src/containers/guest-home-page/signup-dialog/SignupDialog.jsx:90]: I guess useMemo is redundant here

File: src/containers/guest-home-page/signup-dialog/SignupDialog.jsx [hunk 6]
```diff
@@ -84,7 +85,24 @@ const SignupDialog = ({ type }) => {
...
+  const buttonWidths = {
+    small: 330,
+    medium: 333,
+    large: 340,
+    xLarge: 394
+  }
+
+  const getButtonWidth = () => {
+    if (isSmallScreen) return buttonWidths.small
+    if (isMediumScreen) return buttonWidths.medium
```
Comment (Tolik170) [inline src/containers/guest-home-page/signup-dialog/SignupDialog.jsx:102]: maybe it is better to make this function accept a breakpoints object and restructure it inside:

const breakpoints = useBreakpoints()

then, pass breakpoints to this function and restructure it there, also buttonWidth seems redundant, just return the width immidiately

File: src/containers/guest-home-page/login-dialog/LoginDialog.styles.js (hunk 1 above)
Comment (Mygal-Alex) [reply src/containers/guest-home-page/login-dialog/LoginDialog.styles.js]: fixed

File: src/containers/guest-home-page/signup-dialog/SignupDialog.jsx (hunk 2 above)
Comment (Mygal-Alex) [reply src/containers/guest-home-page/signup-dialog/SignupDialog.jsx]: fixed

File: src/hooks/use-breakpoints-google.tsx (hunk 3 above)
Comment (Mygal-Alex) [reply src/hooks/use-breakpoints-google.tsx]: fixed

File: src/containers/guest-home-page/signup-dialog/SignupDialog.styles.js (hunk 4 above)
Comment (Mygal-Alex) [reply src/containers/guest-home-page/signup-dialog/SignupDialog.styles.js]: fixed

File: src/containers/guest-home-page/signup-dialog/SignupDialog.jsx (hunk 5 above)
Comment (Mygal-Alex) [reply src/containers/guest-home-page/signup-dialog/SignupDialog.jsx]: removed useMemo

File: src/containers/guest-home-page/signup-dialog/SignupDialog.jsx (hunk 6 above)
Comment (Mygal-Alex) [reply src/containers/guest-home-page/signup-dialog/SignupDialog.jsx]: image
this implementation looks better ?

File: src/containers/guest-home-page/login-dialog/LoginDialog.jsx [hunk 7]
```diff
@@ -56,7 +56,7 @@ const LoginDialog = () => {
             handleChange={handleInputChange}
             handleSubmit={handleSubmit}
           />
-          <GoogleLogin buttonWidth={styles.form.maxWidth} type={login} />
+          <GoogleLogin buttonWidth={339} type={login} />
```
Comment (Tolik170) [inline src/containers/guest-home-page/login-dialog/LoginDialog.jsx:59]: guess, you shouldn`t remove styles.form.maxWidth, just change the width in styles

File: src/containers/guest-home-page/signup-dialog/SignupDialog.jsx (hunk 6 above)
Comment (Tolik170) [reply src/containers/guest-home-page/signup-dialog/SignupDialog.jsx]: yeah, but move useBreakpoints to the top as hooks should be called in the top of a component

**Review by Tolik170 — APPROVED** [review]

File: src/containers/guest-home-page/login-dialog/LoginDialog.jsx (hunk 7 above)
Comment (Mygal-Alex) [reply src/containers/guest-home-page/login-dialog/LoginDialog.jsx:59]: fixed

**Review by Mav-Ivan — APPROVED** [review]

File: src/containers/guest-home-page/signup-dialog/SignupDialog.jsx (hunk 6 above)
Comment (Mygal-Alex) [reply src/containers/guest-home-page/signup-dialog/SignupDialog.jsx]: image
fixed

**Review by dmtrth25 — APPROVED** [review]

**Review by Marichka0406 — APPROVED** [review]

**Review by dmtrth25 — APPROVED** [review]

**Pull request merged**